### PR TITLE
Refine librispeech.py for DeepSpeech2.

### LINKED ISSUE #77: Refine librispeech.py for DeepSpeech2
1. Add manifest line check.
2. Avoid re-unpacking if unpacked data already exists.
3. Add full_download (download all 7 sub-datasets of LibriSpeech).

### PR DESCRIPTION
**Pull request by xinghai-sun**
resolve #77 

Summary:
1. Add manifest line check.
2. Avoid re-unpacking if unpacked data already exists.
3. Add full_download (download all 7 sub-datasets of LibriSpeech).

### CODE REVIEW

File: deep_speech_2/README.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ For some machines, we also need to install libsndfile1. Details to be added.
 ```
 cd data
 python librispeech.py
+cat manifest.libri.train-* > manifest.libri.train-all
```
Comment (pkuyym) [inline deep_speech_2/README.md:21]: Should we introduce the meaning of manifest.libri.train-* file ?
I see, the introduction details is in following section. Feel abrupt about manifest file.

File: deep_speech_2/data/librispeech.py [hunk 2]
```diff
@@ -27,11 +29,21 @@
...
 
 MD5_TEST_CLEAN = "32fa31d27d2e1cad72775fee3f4849a9"
+MD5_TEST_OTHER = "fb5a50374b501bb3bac4815ee91d3135"
 MD5_DEV_CLEAN = "42e2234ba48799c1f50f24a7926300a1"
+MD5_DEV_OTHER = "c8d0bcc9cca99d4f8b62fcc847357931"
 MD5_TRAIN_CLEAN_100 = "2a93770f6d5c6c964bc36631d331a522"
 MD5_TRAIN_CLEAN_360 = "c0e676e450a7ff2f54aeade5171606fa"
 MD5_TRAIN_OTHER_500 = "d1a0fd59409feb2c614ce4d30c387708"
 
+NUM_LINES_TEST_CLEAN = 2620
```
Comment (pkuyym) [inline deep_speech_2/data/librispeech.py]: Please explain why it's necessary to check the line number when MD5 has been  checked.

**Review by pkuyym — COMMENTED** [review]
Almost LGTM. Please explain the reason of checking the line number when md5 has been checked.

File: deep_speech_2/data/librispeech.py (hunk 2 above)
Comment (kuke) [inline deep_speech_2/data/librispeech.py]: Also feel confused

Comment (kuke) [reply deep_speech_2/data/librispeech.py]: Also feel confused

File: deep_speech_2/README.md (hunk 1 above)
Comment (xinghai-sun) [reply deep_speech_2/README.md:21]: Done.

File: deep_speech_2/data/librispeech.py (hunk 2 above)
Comment (xinghai-sun) [reply deep_speech_2/data/librispeech.py]: Removed.

**Review by pkuyym — APPROVED** [review]
LGTM

**Pull request merged**